src/status/wallet.nim

```diff
@@ -50,34 +50,70 @@ proc initEvents*(self: WalletModel) =
 proc delete*(self: WalletModel) =
   discard
 
-proc sendTransaction*(self: WalletModel, source, to, assetAddress, value, gas, gasPrice, password: string): RpcResponse =
+proc buildTokenTransaction(self: WalletModel, source, to, assetAddress: EthAddress, value: float, transfer: var Transfer, contract: var Contract, gas = "", gasPrice = ""): EthSend =
+  let token = self.tokens.first("address", $assetAddress)
+  contract = getContract("snt")
+  transfer = Transfer(to: to, value: eth2Wei(value, token["decimals"].getInt))
+  transactions.buildTokenTransaction(source, assetAddress, gas, gasPrice)
+
+proc estimateGas*(self: WalletModel, source, to, value: string): int =
+  var tx = transactions.buildTransaction(
+    parseAddress(source),
+    eth2Wei(parseFloat(value), 18)
+  )
+  tx.to = parseAddress(to).some
+  try:
+    let response = eth.estimateGas(tx)
+    result = fromHex[int](response)
+  except RpcException as e:
+    raise
```

Comment:
Should we raise with a message?

Reply:
I see there is a lot of those in this file, so maybe it was a copy paste or maybe just voluntary, I'll let you check

Reply:
The message will get passed through in the `RpcException`